expression/builtin_miscellaneous.go

```diff
@@ -1045,3 +1048,78 @@ type uuidShortFunctionClass struct {
 func (c *uuidShortFunctionClass) getFunction(ctx sessionctx.Context, args []Expression) (builtinFunc, error) {
 	return nil, errFunctionNotExists.GenWithStackByArgs("FUNCTION", "UUID_SHORT")
 }
+
+type vitessHashFunctionClass struct {
+	baseFunctionClass
+}
+
+func (c *vitessHashFunctionClass) getFunction(ctx sessionctx.Context, args []Expression) (builtinFunc, error) {
+	if err := c.verifyArgs(args); err != nil {
+		return nil, err
+	}
+	argTp := args[0].GetType().EvalType()
+	bf, err := newBaseBuiltinFuncWithTp(ctx, c.funcName, args, types.ETString, argTp)
+	if err != nil {
+		return nil, err
+	}
+
+	bf.tp.Flen = 8
+	types.SetBinChsClnFlag(bf.tp)
+
+	sig := &builtinVitessHashSig{bf}
+	sig.setPbCode(tipb.ScalarFuncSig_VitessHash)
+	return sig, nil
+}
+
+type builtinVitessHashSig struct {
+	baseBuiltinFunc
+}
+
+func (b *builtinVitessHashSig) Clone() builtinFunc {
+	newSig := &builtinVitessHashSig{}
+	newSig.cloneFrom(&b.baseBuiltinFunc)
+	return newSig
+}
+
+// evalString evals VITESS_HASH(int64|string|decimal).
+func (b *builtinVitessHashSig) evalString(row chunk.Row) (string, bool, error) {
```

Comment:
Should we implement the vectorized expression function for `builtinVitessHashSig` in `expression/builtin_miscellaneous_vec.go`?

Reply:
@Reminiscent Could you explain what are "vectorized expression functions" in this context?

Reply:
> @Reminiscent Could you explain what are "vectorized expression functions" in this context?

You can see this [blog](https://pingcap.com/blog/10x-performance-improvement-for-expression-evaluation-made-possible-by-vectorized-execution) for more information. In short, the expression function  `builtinVitessHashSig.evalString` here processes one row of data at a time. And if you implement the vectorized expression function for `builtinVitessHashSig.*vecEvalString*` in `expression/builtin_miscellaneous_vec.go`, it can process a batch at a time. The blog introduces more information.